api/server/router/network/network_routes.go

```diff
@@ -139,20 +132,22 @@ func (n *networkRouter) getNetwork(ctx context.Context, w http.ResponseWriter, r
 	listByFullName := map[string]types.NetworkResource{}
 	listByPartialID := map[string]types.NetworkResource{}
 
-	nw := n.backend.GetNetworks()
+	// TODO(@cpuguy83): All this logic for figuring out which network to return does not belong here
+	// Instead there should be a backend function to just get one network.
+	nw := n.backend.GetNetworks(types.NetworkListConfig{Detailed: true, Verbose: verbose})
```

Comment:
Should this always use `Detailed`? Looking at this, this will fetch _all_ networks first (without filter), and thus also _all_ containers for all networks, and then throw away the networks that we don't need.

Wondering if this is a concern, because the reason the detailed response was disabled by default is to prevent fetching all those resources

Perhaps we can;

- move filtering to the backend (?)
- if that's not possible, fetch all networks without details
- once we found the right network, fetch details using that network's ID

Reply:
Aha, I knew there was a reason I'd moved filtering at first.
Will update.

Reply:
Updated